components/camel-qdrant/src/main/java/org/apache/camel/component/qdrant/QdrantProducer.java

```diff
@@ -235,6 +238,33 @@ private boolean delete(Exchange exchange, AsyncCallback callback) {
         return false;
     }
 
+    private boolean createCollection(Exchange exchange, AsyncCallback callback) {
+        final Message in = exchange.getMessage();
+        final VectorParams body = in.getBody(VectorParams.class);
+
+        if (body == null) {
+            exchange.setException(new QdrantActionException(
+                    QdrantAction.UPSERT,
+                    "A payload of type VectorParams.class is expected"));
```

Comment:
Guess this  exception must be adapted to reflect the action name  ?

Reply:
Good catch. Fixed!